traffic_ops/traffic_ops_golang/auth/authorize.go

```diff
@@ -63,26 +64,27 @@ type key int
 
 const CurrentUserKey key = iota
 
-// GetCurrentUserFromDB  - returns the id and privilege level of the given user along with the username, or -1 as the id, - as the userName and PrivLevelInvalid if the user doesn't exist.
-func GetCurrentUserFromDB(DB *sqlx.DB, CurrentUserStmt, user string, timeout time.Duration) CurrentUser {
+// GetCurrentUserFromDB  - returns the id and privilege level of the given user along with the username, or -1 as the id, - as the userName and PrivLevelInvalid if the user doesn't exist, along with a user facing error, a system error to log, and an error code to return
+func GetCurrentUserFromDB(DB *sqlx.DB, CurrentUserStmt, user string, timeout time.Duration) (CurrentUser, error, error, int) {
 	var currentUserInfo CurrentUser
 	if DB == nil {
-		log.Errorf("no db provided to GetCurrentUserFromDB")
-		return CurrentUser{"-", -1, PrivLevelInvalid, TenantIDInvalid, []string{}}
+		return CurrentUser{"-", -1, PrivLevelInvalid, TenantIDInvalid, []string{}}, errors.New("internal server error"), errors.New("no db provided to GetCurrentUserFromDB"), http.StatusInternalServerError
 	}
 	dbCtx, dbClose := context.WithTimeout(context.Background(), timeout)
 	defer dbClose()
 
 	err := DB.GetContext(dbCtx, &currentUserInfo, CurrentUserStmt, user)
 	switch {
 	case err == sql.ErrNoRows:
-		log.Errorf("checking user %v info: user not in database", user)
-		return CurrentUser{"-", -1, PrivLevelInvalid, TenantIDInvalid, []string{}}
+		return CurrentUser{"-", -1, PrivLevelInvalid, TenantIDInvalid, []string{}}, errors.New("user not found"), errors.New(fmt.Sprintf("checking user %v info: user not in database", user)), http.StatusUnauthorized
+	case err == context.DeadlineExceeded || err == context.Canceled:
+		log.Errorf("db access timed out: %s number of open connections: %d\n", err, DB.Stats().OpenConnections)
+		return CurrentUser{"-", -1, PrivLevelInvalid, TenantIDInvalid, []string{}}, errors.New("service unavailable"), err, http.StatusServiceUnavailable
 	case err != nil:
 		log.Errorf("Error checking user %v info: %v", user, err.Error())
-		return CurrentUser{"-", -1, PrivLevelInvalid, TenantIDInvalid, []string{}}
+		return CurrentUser{"-", -1, PrivLevelInvalid, TenantIDInvalid, []string{}}, errors.New("internal server error"), err, http.StatusInternalServerError
```

Comment:
Nitpick: If the caller can use `api.HandleErr`, it'll automatically return a user error of the status code's default text if the user err is nil and system isn't, so this can be `nil, err, http.StatusInternalServerError`